version: "3.9"

services:
  backend:
    image: test_backend
    build:
      context: ./back/backendtest
    ports:
      - "3000:3000"
    expose:
      - "3000"
    environment:
      DATABASE_URL: postgres://admin:admin123@db:5432/mydb 
    depends_on:
      db:
        condition: service_healthy
    networks:
      - dev
    volumes:
      - ./back/backendtest:/app 

  frontend:
    image: test_frontend
    build:
      context: ./front/frontedtest
    environment:  
      VITE_PROXY_API: http://backend:3000/  
    ports:
      - "5173:5173"
    expose:
      - "5173"
    depends_on:
      - backend
    networks:
      - dev

  db:
    image: postgres:16.0
    environment:
      POSTGRES_USER: admin
      POSTGRES_PASSWORD: admin123
      POSTGRES_DB: mydb
    ports:
      - "5432:5432"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U $${POSTGRES_USER} -d $${POSTGRES_DB}"]
      interval: 10s
      timeout: 5s
      retries: 5
    volumes:
      - db-data:/var/lib/postgresql/data
    networks:
      - dev

  nginx:
    image: nginx:stable-alpine
    ports:
      - "80:80"
    volumes:
      - ./nginx.conf:/etc/nginx/nginx.conf
    depends_on:
      - backend
      - frontend
    networks:
      - dev

volumes:
  db-data:

networks:
  dev:
    driver: bridge